src/components/ContactsList/index.js

```diff
@@ -0,0 +1,144 @@
+  },
+  {
+    _id: sampleId++,
+    name: { givenName: "Bertrand", familyName: "Alevin" },
+    avatar: "",
+    phone: [{ number: "06 12345678", primary: true }],
+    email: [{ address: "jeanneige@mail.com", primary: true }]
+  },
+  {
+    _id: sampleId++,
+    name: { givenName: "André", familyName: "Manoukian" },
+    avatar: "",
+    phone: [{ number: "06 12345678", primary: true }],
+    email: [{ address: "jeanneige@mail.com", primary: true }]
+  },
+  {
+    _id: sampleId++,
+    name: { givenName: "Erika", familyName: "Shell" },
+    avatar: "",
+    phone: [{ number: "06 12345678", primary: true }],
+    email: [{ address: "jeanneige@mail.com", primary: true }]
+  },
+  {
+    _id: sampleId++,
+    name: { givenName: "Brendan", familyName: "Abolivier" },
+    avatar: "",
+    phone: [{ number: "06 12345678", primary: true }],
+    email: [{ address: "jeanneige@mail.com", primary: true }]
+  },
+  {
+    _id: sampleId++,
+    name: { givenName: "COZY", familyName: "" },
+    avatar: "",
+    phone: [{ number: "06 12345678", primary: true }],
+    email: [{ address: "jeanneige@mail.com", primary: true }]
+  },
+  {
+    _id: sampleId++,
+    name: { givenName: "EDF", familyName: "" },
+    avatar: "",
+    phone: [{ number: "06 12345678", primary: true }],
+    email: [{ address: "jeanneige@mail.com", primary: true }]
+  },
+  {
+    _id: sampleId++,
+    name: { givenName: "CAF", familyName: "" },
+    avatar: "",
+    phone: [{ number: "06 12345678", primary: true }],
+    email: [{ address: "jeanneige@mail.com", primary: true }]
+  },
+  {
+    _id: sampleId++,
+    name: { givenName: "Eric", familyName: "D" },
+    avatar: "",
+    phone: [{ number: "06 12345678", primary: true }],
+    email: [{ address: "jeanneige@mail.com", primary: true }]
+  }
+];
+
+export const contactSort = (contact, comparedContact) => {
+  const firstNameA =
+    contact.name && contact.name.givenName
+      ? contact.name.givenName.toUpperCase()
+      : "";
+  const firstNameB = comparedContact.name
+    ? comparedContact.name.givenName.toUpperCase()
+    : "";
+
+  if (firstNameB === "") return -1;
+  if (firstNameA === "") return 1;
+  if (firstNameA < firstNameB) return -1;
+
+  if (firstNameA > firstNameB) return 1;
+
+  const lastNameA =
+    contact.name && contact.name.familyName
+      ? contact.name.familyName.toUpperCase()
+      : "";
+  const lastNameB =
+    comparedContact.name && comparedContact.name.familyName
+      ? comparedContact.name.familyName.toUpperCase()
+      : "";
+  if (lastNameA < lastNameB) return -1;
+  if (lastNameA > lastNameB) return 1;
+  return 0;
+};
+
+export const withContacts = BaseComponent =>
+  class WithContacts extends Component {
+    state = {
+      contacts: sampleContacts
+    };
+
+    async componentDidMount() {
+      const contactsRaw = await cozy.client
+        .fetchJSON("GET", "/data/io.cozy.contacts/_all_docs?include_docs=true")
+        .then(data => data.rows)
```

Comment:
Why do you use `await` *and* `.then`?

Reply:
Because what I await is a clean contact list.
I guess cozy-client will return this `contactsRaw` data, so I have not a lot to change when connecting to cozy-client structure.

It is a semantic choice: await some well built data, instead of creating a lot of unecessary variables.